crates/optimism/flashblocks/src/service.rs

```diff
@@ -77,12 +128,15 @@ impl<
     pub fn add_flash_block(&mut self, flashblock: FlashBlock) {
         // Flash block at index zero resets the whole state
         if flashblock.index == 0 {
-            self.blocks = vec![flashblock];
+            self.blocks.clear();
             self.current.take();
+            self.blocks.insert(flashblock);
         }
         // Flash block at the following index adds to the collection and invalidates built block
-        else if flashblock.index == self.blocks.last().map(|last| last.index + 1).unwrap_or(0) {
-            self.blocks.push(flashblock);
+        else if flashblock.index ==
+            self.blocks.iter_ready().last().map(|last| last.index + 1).unwrap_or(0)
+        {
+            self.blocks.insert(flashblock);
             self.current.take();
         }
```

Comment:
is this check then still relevant if we don't care about the order

then all of this can be simplified, but we should check that they belong to the same block